nni/algorithms/compression/pytorch/pruning/mixed_masker_pruner.py

```diff
@@ -0,0 +1,124 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import logging
+from schema import And, Optional
+
+from nni.compression.pytorch.utils.config_validation import CompressorSchema
+from nni.algorithms.compression.pytorch.pruning.one_shot import _StructuredFilterPruner
+
+_logger = logging.getLogger(__name__)
+
+
+class MixedMaskerPruner(_StructuredFilterPruner):
+    """
+    MixedMaskerPruner support config different masker in operation level.
+    """
+
+    def __init__(self, model, config_list, optimizer=None, dependency_aware=False, dummy_input=None, maskers_config_dict=None):
+        """
+        Parameters
+        ----------
+        model : torch.nn.Module
+            Model to be pruned
+        config_list : list
+            Supported keys:
+            - sparsity : This is to specify the sparsity operations to be compressed to.
+            - op_types : See specific pruner introduction more information.
+            - pruning_algo(Optional): A tuple of the type of masker and the args of masker, i.e. ('level', {})
+            - masker_name(Optional): If use maskers_config_dict, the value is the key in maskers_config_dict.
+        optimizer: torch.optim.Optimizer
+            Optimizer used to train model
+        dependency_aware: bool
+            If use dependency aware mode
+        dummy_input: torch.Tensor
+            Required in dependency aware mode
+        maskers_config_dict: dict
+            Reuse pruning_algo value in config_list, key is a custom name of masker and value has the same scheme with pruning_algo in config_list. i.e. {'level_0': ('level', {})}
+        """
+        if maskers_config_dict is None:
+            config_list, maskers_config_dict = self.__convert_config_list(config_list)
+        super().__init__(model, config_list, 'mixed', optimizer=optimizer, dependency_aware=dependency_aware,
+                         dummy_input=dummy_input, maskers_config_dict=maskers_config_dict)
+        _logger.debug('Set MixedMasker successfully.')
+
+    def validate_config(self, model, config_list):
+        """
+        Parameters
+        ----------
+        model : torch.nn.Module
+            Model to be pruned
+        config_list : list
+            List on pruning configs
+        """
+        schema = CompressorSchema([{
+            'sparsity': And(float, lambda n: 0 < n < 1),
+            Optional('op_types'): [str],
+            Optional('op_names'): [str],
+            Optional('masker_name'): str
```

Comment:
"masker_name"? in your example "pruning_algo" is used

Reply:
yes, because we convert config_list, pop "pruning_algo" and add "masker_name"